legal-api/src/legal_api/models/business_common.py

```diff
@@ -187,7 +187,7 @@ def legal_name(self):
         """
         from . import ColinEntity, LegalEntity
 
-        if self.entity_type == self.EntityTypes.SOLE_PROP:
+        if self.is_alternate_name_entity:
```

Comment:
This was causing the below error to occur for GPs:

`AttributeError: 'AlternateName' object has no attribute '_legal_name'. Did you mean: 'legal_name'?`

The reason is that `entity_type` used to [return SP for alternate name entities](https://github.com/bcgov/lear/pull/2535/files#diff-a0b8a49b352e95f68f25b22078477bcbfc2a1d9c34ddc1baab448da949d6dfdcL134), but that functionality was removed when we started storing `entity_type` in the `alternate_names` table.